src/sys/socket/sockopt.rs

```diff
@@ -1424,4 +1551,71 @@ mod test {
         assert!(s_listening2);
         close(s).unwrap();
     }
+
+    #[cfg(any(target_os = "android", target_os = "linux"))]
+    #[test]
+    fn netlink_list_memberships() {
+        use super::super::*;
+        use crate::unistd::close;
+
+        let s = socket(
+            AddressFamily::Netlink,
+            SockType::Raw,
+            SockFlag::empty(),
+            SockProtocol::NetlinkRoute,
+        );
+
+        if let Err(Errno::EPROTONOSUPPORT) = s {
+            return;
+        }
+
+        let s = s.unwrap();
+
+        let groups = || -> Result<Vec<u32>> {
+            setsockopt(s, super::NetlinkAddMembership, &22)?;
+            setsockopt(s, super::NetlinkAddMembership, &27)?;
+            getsockopt(s, super::NetlinkListMemberships)
+        }();
+
+        let groups = match groups {
+            Err(Errno::ENOPROTOOPT) | Err(Errno::EOPNOTSUPP) => return,
+            _ => groups.unwrap(),
+        };
+
+        assert!(groups[0] & (1 << 21) != 0);
```

Comment:
Shouldn't you also assert that there aren't any other bits set?

Reply:
ACK